js/charts/clusteredverticalbar.js

```diff
@@ -223,7 +225,13 @@ function clusteredverticalbar() {
                 }
                 var dimension = _dimension[0];
 
-                _filterDimension[dimension] = filterData.map(function (d) {
```

Comment:
changing the array to object is not needed, it won't work. Please revert your change, and additionally, after you add the data to a map, add this 

```
_filterDimension[dimension]._meta = { 
    dataType: _dimensionType[0],
    valueType: 'castValueType'
};
```

Reply:
I think we need to convert array to object.
because if you want to store values and _data in the same kay so we need to create an object.

Reply:
could you please elaborate why do you think it won't work like this? This has been implemented already a few months ago when I was working on date ranges, not ideal, but seemed like the most optimal way to achieve what we want. If you think that we should change the structure of filter service, then it might lead to huge changes in flair bi app, and can lead to new bugs. 

So in other words, what I am proposing is already a working feature, it's been there for some time, so we just need (with your help) to have vizualizations to start using this feature, that's pretty much it